packages/jest-config/src/readConfigFileAndSetRootDir.ts

```diff
@@ -35,33 +34,10 @@ export default async function readConfigFileAndSetRootDir(
     if (isTS) {
       configObject = await loadTSConfigFile(configPath);
     } else {
-      configObject = require(configPath);
+      configObject = await requireOrImportModule<any>(configPath);
     }
   } catch (error) {
-    if (error.code === 'ERR_REQUIRE_ESM') {
-      try {
-        const configUrl = pathToFileURL(configPath);
-
-        // node `import()` supports URL, but TypeScript doesn't know that
-        const importedConfig = await import(configUrl.href);
-
-        if (!importedConfig.default) {
-          throw new Error(
-            `Jest: Failed to load mjs config file ${configPath} - did you use a default export?`,
-          );
-        }
-
-        configObject = importedConfig.default;
-      } catch (innerError) {
-        if (innerError.message === 'Not supported') {
-          throw new Error(
-            `Jest: Your version of Node does not support dynamic import - please enable it or use a .cjs file extension for file ${configPath}`,
-          );
-        }
```

Comment:
All LTS Node should support dynamic import after Node 10 EOL.

Reply:
Yep. Jest 27 will support Node 10, but 28 will drop it

Reply:
so we should keep the warning for now